app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java

```diff
@@ -72,49 +73,52 @@ public void onClick(View v) {
                 startActivity(webIntent);
             }
         });
 
     }
-    //using check self permission method of ContextCompact checks whether permission is granted or not
+
+    //checks for camera permission
+    //using checkSelfPermission method of ContextCompact checks whether permission is granted or not
     private void askCameraPermissions() {
-        //PERMISSION_GRANTED gives binary output if is it equal to PackageManager.PERMISSION_GRANTED then the user has already given the access
+        //checks for permission from manifest file using PackageManager.PERMISSION_GRANTED
         if(ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED){
-            //Activitycompat class requests permission during runtime
-            //Permission is passed within the string from manifest
+            //Activitycompat class requestPermissions method requests permission during runtime
+            //permission for camera is passed within the string
             ActivityCompat.requestPermissions(this, new String[] {Manifest.permission.CAMERA}, CAMERA_PERM_CODE);
         }else {
-            dispatchTakePictureIntent();//directs to dispatchTakePictureIntent method
+            //user has given permission to the app
+            dispatchTakePictureIntent();  //directs to dispatchTakePictureIntent
         }
     }
     
     //if camera permissions are not given override the askCameraPermission
     @Override
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-        //checking whether permission is given for camera by comparing request code of camera with request code passed to onRequestPermissionResult
+        //checking  permission  by comparing request code of camera with request code passed to onRequestPermissionResult
         if (requestCode == CAMERA_PERM_CODE) {
             //checking grantResults array is empty
-            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED){//if both the conditions are true then camera permission is given
+            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED){  //camera permission given
                 dispatchTakePictureIntent();
-            } else {//toast message appear when both conditions are false
+            } else {
+                //toast message appear when both conditions are false
                 Toast.makeText(this, "Camera Permission is Required to Use camera.", Toast.LENGTH_SHORT).show();
             }
         }
     }
-    // onActivityResult method is used to display and save image
+    // onActivityResult method is used to display and save image as data of this app
```

Comment:
either provide a blank space after a comment or don't. But follow this style throughout the project!